### ISSUE TITLE
Explore using amp-state to compute a score from several quiz answers

### ISSUE THREAD

**Issue by gmajoulet** [ampproject/amphtml#28060]
Quizzes are standalone components: you can create a 10 quizzes story and users answer each one of them individually, but can't get a "score" like "You got 8 questions right!".

We should explore using `amp-state` so publishers can compute a score from several quiz answers.

cc @ampproject/wg-stories @mszylkowski

**Comment by newmuis** [comment]
Note that we could also implement this as a quiz-specofic component if we wanted.

There are other potentially interesting stateful things we could do with amp-state, but maybe they require more thinking.  I think we should explore both, but just wanted to call out that we have the option of just doing something wuiz- or reaction-specific.

**Comment by mszylkowski** [comment]
Definitely, if we wanted to do it on the frontend side, it would require an important setup but wouldn't be impossible with the current quiz/poll implementation. I could draft a "results" component without much thought and revise it. But we might want to lock page advancement until the quiz is responded for example, or not show the results page until all quizzes are answered.

Also, a "results" page could be implemented for polls similar to "what animal/character are you?" quizzes, where the results of the polls get aggregated in a similar style to Buzzfeed quizzes [(example)](https://www.buzzfeed.com/hayleyrochelletillett/agree-disagree-mean-girls-test). I think there's value in that format but might be more complex.

**Comment by mszylkowski** [comment]
Status update: Using amp-bind and AMP.setState internally we can generate bindings with the desired properties that can be used on the story as needed. Currently on experimentation phase.
